### DISCUSSION TITLE
difficulties to boot windows guest

### DISCUSSION THREAD

**Discussion by Y-hwan** [cloud-hypervisor/cloud-hypervisor#3644] [Q&A]
I'm trying to boot a windows guest, but having some difficulties.

1. CLOUDHV.fd couldn't use with QEMU

   Refer to https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/uefi.md, I download [https://github.com/cloud-hypervisor/edk2/releases/download/ch-cad5f05068/CLOUDHV.fd]. It couldn't use with QEMU. show "guest has not initialized the display(yet)". [https://github.com/cloud-hypervisor/edk2/releases/download/ch-40fbc7e24c/OVMF.fd] is fine.
   qemu: QEMU emulator version 6.0.0
   kernel：4.18.0-193

`   ...qemu-system-x86_64 -machine q35,accel=kvm -cpu host -m 4G -bios  /test/cloud-hypervisor/cloudhv.fd  -cdrom /test/base/cn_windows_server_2016_x64_dvd_9718765.iso -drive file=/test/base/virtio-win-0.1.185.iso,index=0,media=cdrom -drive if=none,id=root,file=/test/base/qemu-ch.qcow2 -device virtio-blk-pci,drive=root,disable-legacy=on -device virtio-net-pci,netdev=mynet0,disable-legacy=on -netdev tap,id=mynet0,ifname=ich4,script=no,downscript=no -vnc 0.0.0.0:10 -vga std`

**Comment by Y-hwan** [comment]
2. windows images seems to work fail
    I install window with qemu, and make sure net and RDP working. Then i boot a windows image with cloud-hypervisor. I can not connect to windows.

    command:
     `cloud-hypervisor --kernel /test/cloud-hypervisor/ovmf.fd --disk path=/test/base/win-ovmf --cpus boot=4,kvm_hyperv=on --memory size=4G --net tap=ich4,mac=2e:ee:44:44:17:f3,mask=255.255.255.0,num_queues=2 --serial tty --console off --seccomp false`

    cloud-hypervisor v20.1.0:
    serial output:
`ProtectUefiImageCommon - 0xBE36D040
  - 0x0000000010000000 - 0x000000000015F000
!!!! X64 Exception Type - 0E(#PF - Page-Fault)  CPU Apic ID - 00000000 !!!!
ExceptionData - 0000000000000009  I:0 R:1 U:0 W:0 P:1 PK:0 SS:0 SGX:0
RIP  - 00000000BF8BF43C, CS  - 0000000000000038, RFLAGS - 0000000000010206
RAX  - 00000000BF7EEB98, RCX - 00000000BFD2A4C0, RDX - 0000000000000000
RBX  - 00000000BFD2A4C0, RSP - 00000000BFD2A430, RBP - 00000000BFD2A559
RSI  - 0000000000000000, RDI - 00000000BFD2A4C0
R8   - 000000000016D9B0, R9  - 0000000000399420, R10 - 0000000000000000
R11  - 00000000BFD2A4D0, R12 - 8000000000000002, R13 - 0000000000170A70
R14  - 0000000000000008, R15 - 0000000000000001
DS   - 0000000000000030, ES  - 0000000000000030, FS  - 0000000000000030
GS   - 0000000000000030, SS  - 0000000000000030
CR0  - 0000000080010033, CR2 - 00000000BF8C2128, CR3 - 00000000BFA01000
CR4  - 0000000000040668, CR8 - 0000000000000000
DR0  - 0000000000000000, DR1 - 0000000000000000, DR2 - 0000000000000000
DR3  - 0000000000000000, DR6 - 00000000FFFF0FF0, DR7 - 0000000000000400
GDTR - 00000000BF7DE000 0000000000000047, LDTR - 0000000000000000
IDTR - 00000000BF39A018 0000000000000FFF,   TR - 0000000000000000
FXSAVE_STATE - 00000000BFD2A090`

   then I use cloud-hypervisor v17.0, show "Error booting VM: VmBoot(KernelLoad(Elf(InvalidElfMagicNumber)))"

**Reply by iggy** [reply]
I'm assuming you converted the disk image used in qemu to raw format before you tried to boot it in cloud-hypervisor? I don't think qcow images are officially supported.

**Reply by rbradford** [reply]
Why are you using v17.0? The latest release is v21.0. Further I don't think you can use the new CLOUDHV.fd with QEMU you should use a classic OVMF.fd.

**Reply by Y-hwan** [reply]
Thinks, changing to raw format and using classic OVMF.fd resolve my problem. Linux images with qcow format can boot normally, so I didn't notice this.

**Comment by sboeuf** [comment]
@Y-hwan You simply can't boot QEMU with `CLOUDHV.fd`. The `CloudHvX64` target is dedicated to Cloud Hypervisor.

**Reply by rbradford** [reply]
I think that the documentation implies that this works...because it used to.

**Reply by sboeuf** [reply]
Well we need to update it then :)

**Comment by Y-hwan** [comment]
Can someone tell me why qcow2 don't support windows images? what should I do if I want to implement it?

**Reply by rbradford** [reply]
I think it should work? Does it fail when you try it?

**Reply by sboeuf** [reply]
Might be compressed QCOW the issue?

**Reply by rbradford** [reply]
In the earlier comment @Y-hwan said they changed both the firmware and the disk format at the same time. The disk format is transparent to the guest. But as @sboeuf points out we don't support QCOW files with compressed blocks.

**Reply by maocili** [reply]
I have same problem I can't boot windows images of qcow2 format. And I check my qcow2, I am just install windows in qemu for qcow2 . What should I do if I want to implement it?

```Image type detected as qcow2
Module nbd is loaded!
mount: /mnt/clh-img-check/: can't find in /etc/fstab.
grep: /mnt/clh-img-check/boot/config*: no such file or directory
VIRTIO not found
umount: /mnt/clh-img-check/: not mounted.
  VirtIO module not found in the image
```

**Reply by rbradford** [reply]
That script only works for Linux guest images.
